common/dbconfig.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "common/config.h"
+#include "common/jsoncpp/include/json/json.h"
+
+/**
+ * {
+ *   "dataset" : {
+ *      // dataset level configs
+ *        "ad_query_join" : {
+ *              "ack_mode" : 1,
+ *       }
+ *    }
+ * }
+ */
+
+namespace common {
+
+struct DBConfig {
+  std::unordered_map<std::string, Config> dataSet;
+};
+
+class DBConfigManager {
+public:
+  static DBConfigManager* get();
+  bool loadJsonObject(const Json::Value& root);
```

Comment:
it seems `loadJsonObject` is only a util function used by `loadJsonStr`, do we need to make it public?

Reply:
+1. the two `load` APIs private should suffice

Reply:
`loadJsonStr` is for testing and for future use. `loadJsonObject` is what will be used from serviceconfig

Reply:
I don't see unit testing for `loadJsonObject`, we could either make this private for now (can always make it public when needed), or add unit testing for any public APIs.

Reply:
Also, can you add comments for these APIs

Reply:
Done ..